### PR TITLE
Enable pip package listing

### LINKED ISSUE #149: Enable pip package listing
**Describe the Feature**
Tern currently only lists OS level packages. It should also list language level packages like pip

**Use Cases**
For containers with python apps

**Implementation Changes**
Add `pip` binary in `command_lib/base.yml`. Then add under it use `pip-licenses` to list pip licenses. This requires pip-licenses to be installed.

### PR DESCRIPTION
**Pull request by rnjudge**
This commit adds commands to base.yml to collect pip package, version,
license and proj_url information using the pip package manager.

When the pip package manager does not have information for a package
license the license text is displayed as UNKNOWN. Unknown licenses will
still be reported in the default report for the sake of transparency.

Changes in snippets.yml were adapted from changes suggested by
abhaykatheria.

Resolves #149

Signed-off-by: Rose Judge <rjudge@vmware.com>
Signed-off by: abhaykatheria <abhay.katheria1998@gmail.com>

### CODE REVIEW

**Comment by rnjudge** [comment]
This PR works for a majority of my testing. The one use case I am running in to trouble with is the ternd docker image built from the Dockerfile in the repo. The `pip3 install --upgrade pip` command in the Dockerfile is creating a pip whiteout file that gets picked up when collecting the list of installed packages in the container the first time around listing packages, but not when the listing command is run a second time to collect version/license/proj_url information. Since the list of packages names collected for `names` the first time around is not the same length as the package names collected for `licenses`, `proj_url` or `versions` the second time around, there are inconsistent lengths and tern ignores the values found. In the terminal output you can see:

```
2020-04-22 15:45:01,675 - DEBUG - rootfs - Running command: sudo unshare -pf --mount-proc=/home/rjudge/.tern/temp/mergedir/proc chroot /home/rjudge/.tern/temp/mergedir /bin/bash -c pkgs=`pip3 list --format=freeze 2> /dev/null | cut -f1 -d'='` && for p in $pkgs; do pip3 show $p 2> /dev/null | head -7 | tail -1 | cut -f2 -d' '; done
2020-04-22 15:45:05,033 - DEBUG - rootfs - Running command: sudo unshare -pf --mount-proc=/home/rjudge/.tern/temp/mergedir/proc chroot /home/rjudge/.tern/temp/mergedir /bin/bash -c pkgs=`pip3 list --format=freeze 2> /dev/null | cut -f1 -d'='` && for p in $pkgs; do pip3 show $p 2> /dev/null | head -7 | tail -1 | cut -f2 -d' '; done
2020-04-22 15:45:08,333 - DEBUG - rootfs - Running command: sudo unshare -pf --mount-proc=/home/rjudge/.tern/temp/mergedir/proc chroot /home/rjudge/.tern/temp/mergedir /bin/bash -c pkgs=`pip3 list --format=freeze 2> /dev/null | cut -f1 -d'='` && for p in $pkgs; do pip3 show $p 2> /dev/null | head -2 | tail -1 | cut -f2 -d' '; done
2020-04-22 15:45:12,254 - DEBUG - rootfs - Running command: sudo unshare -pf --mount-proc=/home/rjudge/.tern/temp/mergedir/proc chroot /home/rjudge/.tern/temp/mergedir /bin/bash -c pip3 list --format=freeze 2> /dev/null | cut -f1 -d'='
2020-04-22 15:45:12,913 - WARNING - common - Inconsistent lengths for key: versions
2020-04-22 15:45:12,913 - WARNING - common - Inconsistent lengths for key: licenses
2020-04-22 15:45:12,913 - WARNING - common - Inconsistent lengths for key: proj_urls
```

even though the same `pip3 list --format=freeze 2> /dev/null | cut -f1 -d'='` command is run to collect the list of packages. This then yields the following incorrect output (notice the mysterious `.wh.pip` whiteout package found):

```
File licenses found in Layer:  None
        Packages found in Layer:  .wh.pip-, certifi-, chardet-, docker-, dockerfile-parse-, idna-, pbr-, pip-, PyYAML-, requests-, setuptools-, six-, stevedore-, tern-, urllib3-, websocket-client-
        Licenses found in Layer:  None
```

This DOESN'T happen in the following scenarios:
1) If I change the `RUN pip3 install --upgrade pip && pip3 install tern` line to `RUN pip3 install tern` -- the correct output is reported.
2) I combine the two run commands into one in the Dockerfile. i.e. `RUN tdnf remove -y toybox && tdnf install -y tar findutils attr util-linux python3 python3-pip python3-setuptools git && pip3 install --upgrade pip && pip3 install tern`

**Comment by nishakm** [comment]
#658 should address the whiteout issue

File: tern/formats/default/generator.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 # global logger
 logger = logging.getLogger(constants.logger_name)
 
+# Licenses not to be included in the final report
+ignore_list = ['UNKNOWN', 'unknown', 'Unknown']
```
Comment (nishakm) [inline tern/formats/default/generator.py:22]: I would rather leave anything reported as "unknown" even if it is added by default.

Comment (rnjudge) [reply tern/formats/default/generator.py:22]: Ok, this means that `UNKNOWN` will show up in the list of licenses in the default report.

Comment (rnjudge) [reply tern/formats/default/generator.py]: PR and commit message is updated :)

**Pull request merged**